src/main/webapp/app/components/tabs/CurationDataImportTab.tsx

```diff
@@ -0,0 +1,449 @@
+import { componentInject } from 'app/shared/util/typed-inject';
+import { IRootStore } from 'app/stores';
+import React, { useEffect, useState } from 'react';
+import { Alert, Col, Input, Row } from 'reactstrap';
+import Select, { GroupBase, OptionsOrGroups } from 'react-select';
+import OncoKBTable from 'app/shared/table/OncoKBTable';
+import { filterByKeyword } from 'app/shared/util/utils';
+import { AsyncSaveButton } from 'app/shared/button/AsyncSaveButton';
+import { FirebaseGeneService } from 'app/service/firebase/firebase-gene-service';
+import { getFirebaseGenePath } from 'app/shared/util/firebase/firebase-utils';
+import { GenomicIndicator } from 'app/shared/model/firebase/firebase.model';
+import { observer } from 'mobx-react';
+import StatusIcon, { IconType, Status } from 'app/shared/icons/StatusIcon';
+import { FirebaseGeneReviewService } from 'app/service/firebase/firebase-gene-review-service';
+
+export interface ICurationToolsTabProps extends StoreProps {}
+
+type ImportStatus = 'initialized' | 'pending' | 'complete' | 'warning' | 'error';
+
+type DataImportStatus = {
+  status: ImportStatus;
+  message: string;
+};
+
+type DataRow = {
+  data: { [key: string]: string };
+} & DataImportStatus;
+
+enum DataImportType {
+  GENE_GENOMIC_INDICATOR,
+  GENE_SUMMARY,
+  GENE_BACKGROUND,
+}
+
+const getSelectOption = (dataImportType: DataImportType) => {
+  return {
+    value: dataImportType,
+    label: dataImportTypesLabel[dataImportType],
+  };
+};
+const getSelectOptions = (isGermline: boolean): OptionsOrGroups<any, GroupBase<any>> => {
+  if (isGermline) {
+    return [
+      {
+        label: 'Gene Level',
+        options: [DataImportType.GENE_SUMMARY, DataImportType.GENE_BACKGROUND, DataImportType.GENE_GENOMIC_INDICATOR].map(getSelectOption),
+      },
+    ];
+  } else {
+    return [
+      {
+        label: 'Gene Level',
+        options: [DataImportType.GENE_SUMMARY, DataImportType.GENE_BACKGROUND].map(getSelectOption),
+      },
+    ];
+  }
+};
+const dataImportTypesLabel: { [key in DataImportType]: string } = {
+  [DataImportType.GENE_GENOMIC_INDICATOR]: 'Genomic Indicator',
+  [DataImportType.GENE_SUMMARY]: 'Summary',
+  [DataImportType.GENE_BACKGROUND]: 'Background',
+};
+
+const getRequiredColumns = (dataType: DataImportType) => {
+  switch (dataType) {
+    case DataImportType.GENE_GENOMIC_INDICATOR:
+      return ['hugo_symbol', 'genomic_indicator'];
+    case DataImportType.GENE_SUMMARY:
+    case DataImportType.GENE_BACKGROUND:
+      return ['hugo_symbol', 'value'];
+    default:
+      return [];
+  }
+};
+
+const getOptionalColumns = (dataType: DataImportType) => {
+  switch (dataType) {
+    case DataImportType.GENE_GENOMIC_INDICATOR:
+      return ['entrez_gene_id', 'description', 'alleleState'];
```

Comment:
I don't see `entrez_gene_id` or `alleleState` being used elsewhere. Do we currently plan on supporting importing these fields?

Reply:
I will get rid of entrez_gene_id. My intention was to avoid outdated hugo symbol. But currently I'm not planning to make an API to server side to verify the hugo symbol. The hugo symbol has to be the latest one.

Reply:
`allele_state` I will support